integration-tests/opentelemetry-reactive/src/main/java/io/quarkus/it/opentelemetry/reactive/EndUserResource.java

```diff
@@ -0,0 +1,109 @@
+package io.quarkus.it.opentelemetry.reactive;
+
+import jakarta.annotation.security.PermitAll;
+import jakarta.annotation.security.RolesAllowed;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.semconv.SemanticAttributes;
+
+@Path("/otel/enduser")
+public class EndUserResource {
+
+    @Inject
+    Tracer tracer;
+
+    @Path("/no-authorization")
+    @GET
+    public String noAuthorization() {
+        return "/no-authorization";
+    }
+
+    @RolesAllowed("WRITER")
+    @Path("/roles-allowed-only")
+    @GET
+    public String rolesAllowedOnly() {
+        return "/roles-allowed-only";
```

Comment:
Could you add the list of roles to the endpoint name?... for clarity.
This would turn the path and the return into `roles-allowed-only--writer` or something similar.